h/api/models.py

```diff
@@ -166,6 +176,76 @@ def title(self):
             return ""
 
     @property
+    def filename(self):
+        if self.uri and self.uri.startswith("file://"):
+            return self.uri.split("/")[-1] or None
+
+    @property
+    def document_link(self):
+        """Return a link to this annotation's document.
```

Comment:
Yeah, that would simplify things :) Main reason I'm doing this in Python instead of in Jinja2 is that the logic is complex enough that I want unit tests for it. But if we can put the `<a>` tag in either way and the only logic Jinja2 needs is to output the `href` only if there's something to link to, that should be fine